File: cpp/268.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int size = nums.size();
        for (int i = 0; i < size; ++i) {
            while (nums[i] < size && nums[nums[i]] != nums[i])
                swap(nums[nums[i]], nums[i]);
        }
        for (int i = 0; i < size; ++i)
            if (nums[i] != i)
                return i;
        return size;
    }
};

File: go/268.go
func missingNumber(nums []int) int {
    size := len(nums)
    for i := 0; i < size; i++ {
        for nums[i] < size && nums[nums[i]] != nums[i] {
            nums[i], nums[nums[i]] = nums[nums[i]], nums[i]
        }
    }
    for i := 0; i < size; i++ {
        if i != nums[i] {
            return i
        }
    }
    return size
}
